python/ray/serve/controller.py

```diff
@@ -420,13 +436,46 @@ def deploy_app(
                 "previous request."
             )
 
-        deployment_override_options = config.dict(
-            by_alias=True, exclude_unset=True
-        ).get("deployments", [])
-
         self.config_deployment_request_ref = run_graph.options(
             runtime_env=config.runtime_env
-        ).remote(config.import_path, config.runtime_env, deployment_override_options)
+        ).remote(
+            config.import_path,
+            config.runtime_env,
+            deployment_override_options,
+            version_dict,
+        )
+
+    def get_version_of_new_deployments(
+        self,
+        new_deployments: List[Dict],
+        old_deployments: List[Dict],
+        version_dict: Dict,
+    ):
```

Comment:
```suggestion
    ) -> Dict[str, str]:
```

Make sure to add a return type here.